reaction_cime/migration/versions/20230421_1037_20b7a0330880_add_file_status_field.py

```diff
@@ -0,0 +1,45 @@
+"""add_file_status_field
+
+Revision ID: 20b7a0330880
+Revises: 027d77624694
+Create Date: 2023-04-21 10:37:21.857830
+
+"""
+from alembic import op
+
+
+# revision identifiers, used by Alembic.
+revision = '20b7a0330880'
+down_revision = '027d77624694'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    connection = op.get_bind()
+
+    connection.execute(
+        """
+        ALTER TABLE cime4r.project
+        ADD "file_status" TEXT NOT NULL;
+
```

Comment:
This statement will fail when we deploy it. We have quite a few existing projects.

The best way would be to add the column as nullable, then run a set file_status = done on all rows where it is null (or just all), and then add the not null again.

Reply:
Good catch, with "add the not null again" you mean add the constraint on the column then?

Reply:
@puehringer I think

ALTER TABLE cime4r.project
        ADD "file_status" TEXT NOT NULL DEFAULT 'done';

also does the trick